docs/documentation-principles.md

```diff
@@ -0,0 +1,283 @@
+
+### 1. Document Reality, Not Aspirations
+
+Only document features, behaviors, and guarantees that are actually implemented and
+tested. Documentation should reflect the current state, not planned features.
+
+- Don't document future aspirations as parts of the current feature set or design
+- Don't document incomplete work unless it causes crashes or data corruption
+- If partially implemented, clearly state what works and what doesn't
+- Use issue trackers and PEPs for planned features, not user-facing docs
+
+Good: "Individual operations on `x` are atomic. Sequences of operations require
+external synchronization for deterministic results."
+
+Bad: "The `x` type will be fully thread-safe in a future release."
+
+### 2. Centralize Thread Safety Documentation
+
+Prefer centralized documentation over scattered thread-safety notes. Thread safety
+concerns cut across many APIs; centralizing makes documentation consistent,
+comprehensive, and easier to maintain.
+
+- Create dedicated pages for thread safety guarantees (by type, by module)
+- Link to centralized docs from specific API entries when needed
+- Use module-level notes for most Python APIs, not per-function notes
+- For C APIs, use hybrid approach: centralized overview plus inline notes for
+    critical APIs
+
+Good: Centralized "Thread Safety of Built-in Types" page with per-type guarantees,
+linked from individual type docs.
+
+Bad: Scattered thread safety notes on every method without consistent structure.
+
+### 3. Balance Detail with Succinctness
+
+Provide enough detail for users to write correct code without overwhelming them
+with implementation details.
+
+- Focus on guarantees needed to write correct code
+- Link to PEPs and other resources for implementation details
+- Use examples to illustrate complex concepts
+- Python APIs: module-level summaries, not per-API notes
+- C APIs: more detail since extension authors need lower-level understanding
+
+Good: "Individual list operations like `append()` and `pop()` are atomic. Compound
+operations require external synchronization. See PEP 703 for implementation details."
+
+Bad (too brief): "Thread-safe."
+
+Bad (too detailed): "The list implementation uses biased critical sections. Each
+operation acquires locks in the following order..."
+
+### 4. Distinguish Levels of Support
+
+Clearly differentiate between different levels of thread safety. Not all code needs
+to be fully thread-safe; clear distinctions help users understand guarantees.
+
+- Use consistent terminology (see [Terminology](#free-threading-terminology))
+- Document known limitations explicitly
+- Use trove classifiers for package-level support declarations
+
+Good: "Individual `x` operations are atomic and will not cause crashes. Sequences
+of operations may observe inconsistent state without external synchronization.
+Iteration while another thread modifies the `x` may raise RuntimeError."
+
+Bad: "Thread-safe." (no indication of what this means or limitations)
```

Comment:
I think we may want to even go further: Avoid "Thread Safe" except in Section header and document name.